R2Northstar/NorthstarLauncher
# Inital native code for verified mod auto-downloading

## Description

This branch allows client to download a mod archive from the Thunderstore API, and extract included mods in the remote mods folder of the current game profile.

Not all mods can be automatically downloaded, as it would cause *some* security issues, and Thunderstore mod name cannot be deduced from actual mod name: to be eligible to auto-downloading, a mod must appear in the list of verified mods.

Said list and complete mod verification procedure are described here: https://github.com/R2Northstar/VerifiedMods

This branch exposes two commands to test the feature:
* `fetch_verified_mods` retrieves verified mods list from the GitHub organization, and stores it locally;
* `download_mod` does the actual mod downloading/extraction job.

### What this PR is about

* Downloading a mod archive
* Extracting said archive to mods folder

### What this PR is *NOT* about

* Download progress indicator/messages or UI of any kind
* Mod availability to the game (no `reload_mods` invoked)

## Resources

### How to test

* Download and install DLL file from latest CI build;
* Run the `fetch_verified_mods` command;
* Download a mod with `download_mod [mod_name] [mod_version]` (only `Fifty.mp_frostbite 0.0.1` is listed at the time of writing);
* Check your remote mods directory, the downloaded mod should appear there.

### Media

[autodl-pr-opening.webm](https://github.com/R2Northstar/NorthstarLauncher/assets/11993538/e1ce2f81-3b23-460a-a7c9-5cff60b99751)
